packages/gatsby-theme-docs/utils/generate-toc.js

```diff
@@ -19,7 +19,9 @@ function processToC(node, current, toString) {
   switch (node.type) {
     case `paragraph`: {
       current.title = toString(node);
-      current.url = `#${slugger.slug(current.title)}`;
+      const nodeSlug = slugger.slug(current.title);
+      // remove the trailing dash(es) or underscore(s)
+      current.url = `#${nodeSlug.replace(/[-_]+$/, '')}`;
```

Comment:
please re-use the logic of how this is exactly happening in a utility.  The root cause of our problems is that the implementations of generating the ID and the toc are not matching